specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2025-03-01-preview/PreviewAlertRule.json

```diff
@@ -0,0 +1,621 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2025-03-01-preview",
+    "title": "Preview Alert Rule API in Alerts Management Service Resource Provider",
+    "description": "The Preview Alert Rule API provides the ability to retrieve the results of a historical execution of an alert rule"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.AlertsManagement/previewAlertRule": {
+      "post": {
+        "tags": [
+          "PreviewAlertRule"
+        ],
+        "description": "Retrieves the results of a historical execution of an alert rule",
+        "operationId": "PreviewAlertRule",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/RuleGroupNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "ruleType",
+            "in": "query",
+            "required": true,
+            "type": "string",
+            "description": "The query parameter that specifies rule type to be previewed."
+          },
+          {
+            "name": "lookBackPeriod",
+            "in": "query",
+            "required": true,
+            "type": "duration",
+            "description": "The query parameter that specifies the lookback of the preview."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "oneOf": [
+                {
+                  "type": "object",
+                  "properties": {
+                    "ruleType": {
+                      "type": "string",
+                      "enum": ["scheduledQueryRule"]
+                    },
+                    "bodyParam1": {
+                      "$ref": "#/definitions/ScheduledQueryRuleResource"
+                    }
+                  }
+                }
+              ]
```

Comment:
Interesting. I have never seen this before.

Is the intention to support more rule schemas in the future?

I guess as long as its purely an input, that's technically okay....

Reply:
I think its conceptually probably nicer to understand if your 'one of' is also given a 'named definition' that we refer to.

Reply:
(It might make it easier for SDK generators etc?)

Reply:
oops sorry, commenting on old iterations. Its all different now isn't it.